### PR TITLE
Create Github release and upload artifact for each commit

### PR DESCRIPTION
**Pull request by lemmy**
See Assets in my fork https://github.com/lemmy/vscode-tlaplus/releases.  Useful if people want to distribute the .vsix of their fork.

### CODE REVIEW

**Comment by alygin** [comment]
@lemmy, thanks for the PR!

I like the idea of adding the .vsix file to the release assets automatically (I've been doing this work manually so far). But there're two things that bother me a bit:

- The current version number is a part of the file name, and it's hardcoded now (1.5.2). I think we could find a way to set it automatically. Something like `echo "::set-output name=version::$(jq -r .version package.json)"` should work.
- I'm a bit sceptical about the idea of creating a release on every build. It may clutter releases quickly. The current CI pipeline is triggered by every push at the moment, even if it's not the master branch. I'd prefer having a separate pipeline for creating releases (based on `v*`-tags, for instance).

What do you think?

**Comment by lemmy** [comment]
The version number is no longer hard-coded, and it's only building pushes to master.  For https://github.com/tlaplus/tlapm and https://github.com/tlaplus/communitymodules, we create a release for each commit.  For https://github.com/tlaplus/tlaplus, we maintain a pre-release.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: CI
 
-on: [push]
+on:
+  push:
+    branches:    
+      - master
```
Comment (alygin) [inline .github/workflows/ci.yml:6]: But now we lost a pipeline for non-master pushes.
My idea was to leave `ci.yml` as it was, and create a separate pipelines for releases. For instance, `pre-release.yml` that would create pre-releases on every push to master, and `release.yml` that would create releases when manually triggered.

Comment (lemmy) [reply .github/workflows/ci.yml:6]: In that case, can't we just rename the current `ci.yml` to `pre-release.yml` and then revert to the original `ci.yml`? The `release.yml` seems to be mostly orthogonal, no?

Comment (alygin) [reply .github/workflows/ci.yml:6]: I see a couple of differences in the pipelines: In the pre-release and release pipelines we don't need windows in the OS matrix. And in the release pipline the version must be taken from the package.json, not created based on the current date/time. Later, the release pipeline can be expanded to do real publishing to the marketplace and to Open VSX.

**Comment by alygin** [comment]
@lemmy , I've created a [pre-release pipline](https://github.com/alygin/vscode-tlaplus/blob/master/.github/workflows/pre-release.yml) based on your PR.

**Comment by lemmy** [comment]
The CI.yml fails on Ubuntu because of xvfb: https://github.com/lemmy/vscode-tlaplus/actions/runs/543737254

**Comment by alygin** [comment]
Yeah, I'm trying to fix that

**Pull request closed without merging**